rpi/gpio.go

```diff
@@ -82,39 +147,65 @@ func (pi *piPigpio) GetGPIOBcom(bcom int) (bool, error) {
 func (pi *piPigpio) SetGPIOBcom(bcom int, high bool) error {
 	pi.mu.Lock()
 	defer pi.mu.Unlock()
-	if !pi.gpioConfigSet[bcom] {
-		if pi.gpioConfigSet == nil {
-			pi.gpioConfigSet = map[int]bool{}
+
+	// verify we are currently managing this pin via GPIOPinByName or reconfigure
+	pin, ok := pi.gpioPins[bcom]
+	if !ok {
+		return fmt.Errorf("error getting GPIO pin, pin %v not found", bcom)
+	}
+	// configure the pin to be an output if it is not already an output
+	if pin.configuration != GPIOOutput {
+		// first if the pin was configured for pwm, we should turn off the pwm
+		if pin.configuration == GPIOPWM {
+			res := C.set_PWM_dutycycle(pi.piID, C.uint(pin.pin), C.uint(0))
+			if res != 0 {
+				return errors.Errorf("pwm set fail %d", res)
+			}
 		}
-		res := C.set_mode(pi.piID, C.uint(bcom), C.PI_OUTPUT)
+		res := C.set_mode(pi.piID, C.uint(pin.pin), C.PI_OUTPUT)
 		if res != 0 {
 			return rpiutils.ConvertErrorCodeToMessage(int(res), "failed to set mode")
 		}
-		pi.gpioConfigSet[bcom] = true
 	}
 
 	v := 0
 	if high {
 		v = 1
 	}
-	C.gpio_write(pi.piID, C.uint(bcom), C.uint(v))
+	C.gpio_write(pi.piID, C.uint(pin.pin), C.uint(v))
+
 	return nil
 }
 
 func (pi *piPigpio) pwmBcom(bcom int) (float64, error) {
-	res := C.get_PWM_dutycycle(pi.piID, C.uint(bcom))
+	// verify we are currently managing this pin via GPIOPinByName or reconfigure
+	pin, ok := pi.gpioPins[bcom]
+	if !ok {
+		return 0, fmt.Errorf("error getting GPIO pin, pin %v not found", bcom)
+	}
+	if pin.configuration != GPIOPWM {
+		pi.logger.Debugf("pin %v is currently not configured as pwm", bcom)
+		return 0, nil
+	}
```

Comment:
if the pin was not configured as a pwm then I do not return the duty cycle. 

I did this because i was seeing pigpio return negative/incorrect values for the duty cycle when the pin was set as a basic digital I/O